dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -67,392 +67,351 @@
+                .put("size", stats.getSize())
+                .put("count", stats.getDocCount());
         }
-
-        AdminLogger.log(ESIndexResource.class, "deactivateIndex", "Index deactivated: " + indexName);
-    }
-
-    public static long indexDocumentCount(final String indexName) {
-        return APILocator.getContentletIndexAPI().getIndexDocumentCount(indexName);
+        return Response.ok(new ResponseEntityView(builder.build())).build();
     }
-
-    /**
-     * Creates a compressed (zip) index snapshot file.
-     *
-     * @param httpServletRequest request
-     * @param params optional parameters, such as "alias"
-     * @return
-     * @deprecated Use ES Snapshot via ES REST API instead {@see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/6.7/modules-snapshots.html">}.
-     */
+    
+    @CloseDBIfOpened
     @GET
-    @Path("/snapshot/{params:.*}")
-    @Produces({"application/zip", MediaType.APPLICATION_JSON})
-    @Deprecated
-    public Response snapshotIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) {
-
-    	try {
-        	checkArgument(params != null);
-            InitDataObject initDataObject = auth(params,httpServletRequest, httpServletResponse);
-            final User user = initDataObject.getUser();
-            String indexName = this.indexHelper.getIndexNameOrAlias(initDataObject.getParamsMap(),this.indexAPI);
-
-            if(!UtilMethods.isSet(indexName)){
-            	return this.responseUtil.getErrorResponse(httpServletRequest, Response.Status.BAD_REQUEST, user.getLocale(), null, "snapshot.wrong.arguments");
-            }
-
-            if("live".equalsIgnoreCase(indexName) || "working".equalsIgnoreCase(indexName)){
-                IndiciesInfo info = this.indiciesAPI.loadIndicies();
-                if("live".equalsIgnoreCase(indexName)){
-                    indexName = info.getLive();
+    @JSONP
+    @NoCache
+    @Path("/failed")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response downloadRemainingRecordsAsCsv(@Context final HttpServletRequest request,
+                    @Context final HttpServletResponse response) throws DotDataException {
+        auth(request, response);
+        List<Map<String, Object>> results = new ArrayList<>();
+        
+        
+        
+        APILocator.getReindexQueueAPI().getFailedReindexRecords().stream().forEach(row->{
+            Map<String, Object> failure = new HashMap<>();
+            
+            final Contentlet contentlet = Try.of(()->APILocator.getContentletAPI().findContentletByIdentifierAnyLanguage(row.getIdentToIndex()))
+                            .onFailure(e->Logger.warn(ESIndexResource.class, e.getMessage(), e))
+                            .getOrElse(Contentlet::new);
+            Map<String, Object> conMap = Try.of(() -> new ContentletToMapTransformer(contentlet).toMaps().get(0))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'conMap' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)